src/Blocks/components/form/assets/utilities.js

```diff
@@ -1,88 +1,586 @@
+			let expirationDate = new Date(currentStorage.timestamp);
+			expirationDate.setDate(expirationDate.getDate() + parseInt(expiration, 10));
+
+			// Remove expired storage if it exists.
+			if (expirationDate.getTime() < currentStorage.timestamp) {
+				localStorage.removeItem(this.STORAGE_NAME);
+			}
+		}
+
+		// Create new storage if this is the first visit or it was expired.
+		if (this.getLocalStorage() === null) {
+			localStorage.setItem(
+				this.STORAGE_NAME,
+				JSON.stringify(newStorage)
+			);
+			return;
+		}
+
+		// Prepare new output.
+		const output = {
+			...currentStorageFinal,
+			...newStorageFinal,
+		};
+
+		// If output is empty something was wrong here and just bailout.
+		if (Object.keys(output).length === 0) {
+			return;
+		}
+
+		// If nothing has changed bailout.
+		if (JSON.stringify(currentStorageFinal) === JSON.stringify(output)) {
+			return;
+		}
+
+		// Add timestamp to the new output.
+		const finalOutput = {
+			...output,
+			timestamp: newStorage.timestamp,
+		};
+
+		// Update localStorage with the new item.
+		localStorage.setItem(this.STORAGE_NAME, JSON.stringify(finalOutput));
+	};
+
+	// Get local storage value.
+	getLocalStorage = () => {
+		return localStorage.getItem(this.STORAGE_NAME);
+	};
+
+	// Reset form values if the condition is right.
+	resetForm = (element) => {
+		if (this.formResetOnSuccess) {
+			element.reset();
+
+			const formId = element.getAttribute(this.DATA_ATTRIBUTES.formPostId);
+
+			// Unset the choices in the submitted form.
+			if (this.customSelects[formId]) {
+				this.customSelects[formId].forEach((item) => {
+					item.setChoiceByValue('');
+				});
+			}
+
+			// Unset the choices in the submitted form.
+			if (this.customFiles[formId]) {
+				this.customFiles[formId].forEach((item) => {
+					item.removeAllFiles();
+				});
+			}
+
+			const fields = element.querySelectorAll(this.fieldSelector);
+
+			[...fields].forEach((item) => {
+				item.classList.remove(this.SELECTORS.CLASS_FILLED);
+				item.classList.remove(this.SELECTORS.CLASS_ACTIVE);
+				item.classList.remove(this.SELECTORS.CLASS_HAS_ERROR);
+			});
+
+			// Remove focus from last input.
+			document.activeElement.blur();
+
+			// Dispatch event.
+			this.dispatchFormEvent(element, this.EVENTS.AFTER_FORM_SUBMIT_RESET);
+		}
+	};
+
+	// Redirect to url and update url params from from data.
+	redirectToUrl = (element, formData) => {
+		let redirectUrl = element.getAttribute(this.DATA_ATTRIBUTES.successRedirect) ?? '';
+
+		// Replace string templates used for passing data via url.
+		for (var [key, val] of formData.entries()) { // eslint-disable-line no-unused-vars
+			if (typeof val === 'string') {
+				const { value, name } = JSON.parse(val);
+				redirectUrl = redirectUrl.replaceAll(`{${name}}`, encodeURIComponent(value));
+			}
+		}
+
+		// Do the actual redirect after some time.
+		setTimeout(() => {
+			window.location.href = redirectUrl;
+		}, parseInt(this.redirectionTimeout, 10));
+	};
+}
```

Comment:
not sure how all of these utilities are used in forms. Any suggestions?

Reply:
not sure, but first thing that comes to mind is to move the localstorage logic to a separate file.
E.g. You could call the new localstorage from the utils methods and pass the utils instance to the localstorage so it has all the data. It's not ideal because of the tight coupling, but at least it would be some separation of concerns.

Reply:
good idea I will review it tnx

Reply:
I refactored it so that the local storage is in another file

Reply:
done